components/analytics/system_analytics/system_analytics.tsx

```diff
@@ -313,7 +315,7 @@ export default class SystemAnalytics extends React.PureComponent {
                     />
                 }
                 icon='fa-globe'
-                count={stats[StatTypes.TOTAL_PUBLIC_CHANNELS] + stats[StatTypes.TOTAL_PRIVATE_GROUPS]}
+                count={Number(stats[StatTypes.TOTAL_PUBLIC_CHANNELS]) + Number(stats[StatTypes.TOTAL_PRIVATE_GROUPS])}
```

Comment:
since stats can be either a number or a analitycsRow, won't you get a NaN out of it?
this applies to other changes as well, but this might be more problematic as you might end up with NaN+NaN

Reply:
I'm not really sure how to tackle this.

Any suggestions would be appreciated ?

Thanks for your input.

Reply:
I'd probably use a helper function to ensure it extracts the number out of it wether it is one type or the other.

Doing a quick search I found [this post](https://thoughtbot.com/blog/the-case-for-discriminated-union-types-with-typescript) that might help you. But let me know either here or in the comunity server if you want to discuss it further :)

Reply:
So check if it is a number then use that otherwise if it is an analyticsRow extract the value out of that object { name: string, value: number} and use that ?

Also looking at the StatisticCount component it expects the count prop to be a number and if it's not a number it just renders loading message. So i guess it never was using the analyticsrow type... I'm not sure adding those checks makes any difference ?

does that makes sense or am i completely off here.. please let me know.

Reply:
> So check if it is a number then use that otherwise if it is an analyticsRow extract the value out of that object { name: string, value: number} and use that ?

yes, I think that would be the approach here.

> 
> Also looking at the StatisticCount component it expects the count prop to be a number and if it's not a number it just renders loading message. So i guess it never was using the analyticsrow type... I'm not sure adding those checks makes any difference ?

I think the difference would be that the signature `Dictionary<number | AnalyticsRow[]>` allows it to happen, and if it happens it'll work, instead of showing an eternal "loading..." message that might confuse users. originally it was simply an object with no definition and as such no real constraint for what is accepted. What made you choose `Dictionary<number | AnalyticsRow[]>` as the type for stats?

Reply:
I saw it with those types in other components. Also while inspecting it.. data looks similar to that type.

Reply:
I'm fine with it, but then we have to be able to handle the info :)

so your approach would probably be the way to move forward

Reply:
// Helper Function
private checkStat(stat: number | AnalyticsRow[]):number {
    if (typeof stat === 'number') {    
            return stat;    
        }    
        return stat[0].value;    
}

// Usage

count={this.checkStat(stats[StatTypes.TOTAL_PUBLIC_CHANNELS]) + this.checkStat(stats[StatTypes.TOTAL_PRIVATE_GROUPS])}

I came up with this ?

Reply:
yep, that should work!
I'd probably change the name to `getStatValue` or something along the lines to give a more meaningful name.

Reply:
Done